### DISCUSSION TITLE
Quickly generate a Jingle using Text-to-Speech

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6688] [Show and tell]
Sometimes you just quickly need a short jingle with a bumper, and have no speaker available. Or sit hundreds of miles away and must do it on your server.

This is a little bash script that does just that, using a TTS (Text-to-Speech) engine in a nice way. It can:

- generate an MP3 file from either PicoTTS (local) or gtts (Google TTS)
- perform a little sound optimization on the TTS voice (bass/treble correction, compressor/limiter)
- add a starting/ending bumper, using a sound file
- add a nice ID3 tag (fills in Title, Artist, Album, Year, Genre)
- perform ReplayGain calculation and add ReplayGain tags using [loudgain](https://github.com/Moonbase59/loudgain)

Download & docs: https://github.com/Moonbase59/jingle

**Comment by LordHelmchen666** [comment]
Thanks Moonbase59. I am playing around with your script trying to use [Piper](https://github.com/rhasspy/piper) 
Because pico2wave gives horrible results, gTTS i dont want to use (Google). Piper produces pretty good results and is local.
I try to make some random Jingle every minute. e.g. "Now it is ten o clock", "Today is Friday and you listen to..." and so on.

**Reply by LordHelmchen666** [reply]
I got it working with Piper. Sounds quite well.
Here is the script to randomly generate a jingle.
```
#!/bin/bash
d=$(date +%u)  # day of week (1..7); 1 is Monday 
h=$(date +%H)  # hour (00..23)
m=$(date +%M)  # minute (00..59)
day=$(LC_TIME="de_DE.UTF-8" date +%A)  # Full Name of day e.g. Sunday
month=$(LC_TIME="de_DE.UTF-8" date +%B)  # Full Name of month e.g. February
dom=$(date +%d)  # day of month (e.g, 01) 

## Replace some numbers for correct spelling
# Hours
h="${h/00/Null}"
h="${h/01/Ein}"
h="${h/02/Zwei}"
h="${h/03/Drei}"
h="${h/04/Vier}"
h="${h/05/Fünf}"
h="${h/06/Sechs}"
h="${h/07/Sieben}"
h="${h/08/Acht}"
h="${h/09/Neun}"
# Minutes
m="${m/00/''}"
m="${m/01/Eins}"
m="${m/02/Zwei}"
m="${m/03/Drei}"
m="${m/04/Vier}"
m="${m/05/Fünf}"
m="${m/06/Sechs}"
m="${m/07/Sieben}"
m="${m/08/Acht}"
m="${m/09/Neun}"
# day of month
dom="${dom/01/Erste}"
dom="${dom/02/Zweite}"
dom="${dom/03/Dritte}"
dom="${dom/04/Vierte}"
dom="${dom/05/Fünfte}"
dom="${dom/06/Sechste}"
dom="${dom/07/Siebte}"
dom="${dom/08/Achte}"
dom="${dom/09/Neunte}"
dom="${dom/10/Zehnte}"
dom="${dom/11/Elfte}"
dom="${dom/12/Zwölfte}"
dom="${dom/13/Dreizehnte}"
dom="${dom/14/Vierzehnte}"
dom="${dom/15/Fünfzehnte}"
dom="${dom/16/Sechszehnte}"
dom="${dom/17/Siebzehnte}"
dom="${dom/18/Achtzehnte}"
dom="${dom/19/Neunzehnte}"
dom="${dom/20/Zwanzigste}"
dom="${dom/21/Einundzwangsigste}"
dom="${dom/22/Zweiundzwandzigste}"
dom="${dom/23/Dreiundzwandzigste}"
dom="${dom/24/Vierundzwandzigste}"
dom="${dom/25/Fünfundzwandzigste}"
dom="${dom/26/Sechsundzwandzigste}"
dom="${dom/27/Siebenundzwandzigste}"
dom="${dom/28/Achtundzwandzigste}"
dom="${dom/29/Neunundzwandzigste}"
dom="${dom/30/Dreissigste}"
dom="${dom/31/Einunddreissigste}"


array[0]="Sie hören Eis Radio D E. Wir begleiten sie durch den $day"
array[1]="Mit uns durch den $month. Wir wünschen weiterhin gute Unterhaltung bei Eis Radio D E."
array[2]="Es ist $h Uhr $m. Weiterhin viel Spass bei Eis Radio D E"
array[3]="Sie hören Eis Radio D E. Es ist $day $h Uhr $m"
array[4]="Heute ist $day der $dom $month. Es ist $h Uhr $m. Gute Unterhaltung mit Eis Radio D E."

size=${#array[@]}
index=$(($RANDOM % $size))
text=$(echo ${array[$index]})
jingle 'randomjingle' "$text"

```

**Reply by Moonbase59** [reply]
Thanks for sharing! Btw, the [Piper link](https://github.com/rhasspy/piper) above had an extra period at the end.
